x/gamm/keeper/swap_test.go

```diff
@@ -505,8 +505,32 @@ func (suite *KeeperTestSuite) TestInactivePoolFreezeSwaps() {
 	for _, test := range testCases {
 		suite.Run(test.name, func() {
 			// Check swaps
-			_, swapInErr := gammKeeper.SwapExactAmountIn(suite.Ctx, suite.TestAccs[0], test.pool, testCoin, "bar", sdk.ZeroInt(), sdk.ZeroDec())
-			_, swapOutErr := gammKeeper.SwapExactAmountOut(suite.Ctx, suite.TestAccs[0], test.pool, "bar", sdk.NewInt(1000000000000000000), testCoin, sdk.ZeroDec())
+			_, swapInErr := suite.App.SwapRouterKeeper.RouteExactAmountIn(
```

Comment:
Wdyt about moving this whole test to swaprouter?

Reply:
I would also prefer this way, but one major blocker in doing so is that right now we don't have access to be setting pools directly outside the module via `setPool`. But then, we come across the problem of not being able to set an inactive pool. Lmk if you have thoughts for work around for this

Reply:
Makes sense. The only thing I can think of is unexporting the `setPool`. 

However, I think between the choice of unexporting the method and keeping this test in x/gamm, I would prefer the latter.

Let's keep the test here for now, and if we still find this awkward in the future, we can think about this deeper.

I'm going to merge the PR but let me know if you still would like to continue discussing this separately.